pygmt/datasets/tile_map.py

```diff
@@ -0,0 +1,130 @@
+"""
+Function to load raster tile maps from XYZ tile providers, and load as
+:class:`xarray.DataArray`.
+"""
+
+try:
+    import contextily
+except ImportError:
+    contextily = None
+
+import numpy as np
+import xarray as xr
+
+__doctest_requires__ = {("load_tile_map"): ["contextily"]}
+
+
+def load_tile_map(region, zoom="auto", source=None, lonlat=True, wait=0, max_retries=2):
+    """
+    Load a georeferenced raster tile map from XYZ tile providers.
+
+    The tiles that compose the map are merged and georeferenced into an
+    :class:`xarray.DataArray` image with 3 bands (RGB). Note that the returned
+    image is in a Spherical Mercator (EPSG:3857) coordinate reference system.
+
+    Parameters
+    ----------
+    region : list
+        The bounding box of the map in the form of a list [*xmin*, *xmax*,
+        *ymin*, *ymax*]. These coordinates should be in longitude/latitude if
+        ``lonlat=True`` or Spherical Mercator (EPSG:3857) if ``lonlat=False``.
+
+    zoom : int
+        Optional. Level of detail. [Default is 'auto'].
+
+    source : xyzservices.TileProvider or str
+        Optional. The tile source: web tile provider or path to a local file.
+        The web tile provider can be in the form of a
+        :class:`xyzservices.TileProvider` object or a URL. The placeholders for
+        the XYZ in the URL need to be {x}, {y}, {z}, respectively. For local
+        file paths, the file is read with :doc:`rasterio <rasterio:index>` and
+        all bands are loaded into the basemap. IMPORTANT: tiles are assumed to
+        be in the Spherical Mercator projection (EPSG:3857). [Default is
+        ``xyzservices.providers.Stamen.Terrain``, i.e. Stamen Terrain web
+        tiles].
```

Comment:
Currently, `xyzservices.TileProvider` is linked to its API documentation (https://xyzservices.readthedocs.io/en/stable/api.html#xyzservices.TileProvider) which is very technical.

Perhaps we should add a link to https://contextily.readthedocs.io/en/latest/intro_guide.html#Providers so that users can quickly see the list of all available providers?

Reply:
Ok, I've linked to https://contextily.readthedocs.io/en/stable/providers_deepdive.html which seems a bit better (actually mentioned before in https://github.com/GenericMappingTools/pygmt/pull/2125#discussion_r1060274651) and split out the three possible `source` options as bullet points. Done in commit aa2d5fb1ebb45a072a2df6709918e55c2baccfeb. Take a look to see if it's ok.